lib/openapi_parser/schema_validators/base.rb

```diff
@@ -24,9 +24,11 @@ def validate_discriminator_schema(discriminator, value)
       end
       mapping_key = value[discriminator.property_name]
 
-      # TODO: it's allowed to have discriminator without mapping, then we need to lookup discriminator.property_name
+      # it's allowed to have discriminator without mapping, then we need to lookup discriminator.property_name
       # but the format is not the full path, just model name in the components
-      mapping_target = discriminator.mapping[mapping_key]
+      mapping = discriminator.mapping || ->(key) { "#/components/schemas/#{key}" }
+
+      mapping_target = mapping[mapping_key]
```

Comment:
I think almost ok... but I find other bug.

[The specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#discriminator-object) says

    If the discriminator value does not match an implicit or explicit mapping, 
    no schema can be determined and validation SHOULD fail.


So discriminator have implicit and explicit mapping.
And when the value doesn't match both, the validation should be failed.

So we should this step
1. get explicit mapping_target from discriminator.mapping
2. we can't find it, we should use implicit mapping ( `#/components/schemas/#{key}` ) . 
  (When discriminator without mapping or mapping don't have 'mapping_key' we can't find it)
3. we can't resolve schema, we should raise error.

Reply:
@ota42y I have updated PR to meet the spec above.

I am not sure what to do in some edge cases, where there's a matching entry in an explicit mapping whose schema reference is invalid.
Current implementation treats it as an error, and does not fall back to implicit mapping.

Note:
bfa9860 is not related to this PR. Tell me if I should remove this commit from this PR.

Reply:
I think this code is OK 👍 
Sorry, the master branch is broken now...
I'll fix so please wait....

Reply:
I fixed the master branch.
So please remove conflict to merge master

Reply:
@ota42y I have resolved the conflict, and also removed the workaround commit (bfa9860) for broken specs.